core/src/main/java/org/jdbi/v3/ConcreteStatementContext.java

```diff
@@ -89,11 +98,16 @@ public Object getAttribute(String key)
     }
 
     @Override
-    public <T> ResultColumnMapper<T> columnMapperFor(Class<T> type)
+    public ResultColumnMapper<?> columnMapperFor(Type type)
```

Comment:
Can we hold off on this one? We can always add them as default methods. I'm not convinced we'll need them at the level of the library they'll be used.